src/EnergyPlus/HVACHXAssistedCoolingCoil.cc

```diff
@@ -1070,54 +899,42 @@ namespace HVACHXAssistedCoolingCoil {
+                state.dataLoopNodes->Node(thisHXCoil.HXAssistedCoilInletNodeNum).HumRat;
+            state.dataLoopNodes->Node(thisHXCoil.HXExhaustAirInletNodeNum).Enthalpy =
+                state.dataLoopNodes->Node(thisHXCoil.HXAssistedCoilInletNodeNum).Enthalpy;
+            state.dataLoopNodes->Node(thisHXCoil.HXExhaustAirInletNodeNum).MassFlowRate =
+                state.dataLoopNodes->Node(thisHXCoil.HXAssistedCoilInletNodeNum).MassFlowRate;
         }
 
         // Force at least 2 iterations to pass outlet node information
         while ((std::abs(Error) > 0.0005 && Iter <= MaxIter) || Iter < 2) {
 
-            SimHeatRecovery(state,
-                            state.dataHVACAssistedCC->HXAssistedCoil(HXAssistedCoilNum).HeatExchangerName,
-                            FirstHVACIteration,
-                            state.dataHVACAssistedCC->HXAssistedCoil(HXAssistedCoilNum).HeatExchangerIndex,
-                            FanOpMode,
-                            PartLoadRatio,
-                            HXUnitOn,
-                            CompanionCoilIndexNum,
-                            _,
-                            EconomizerFlag,
-                            _,
-                            state.dataHVACAssistedCC->HXAssistedCoil(HXAssistedCoilNum).CoolingCoilType_Num);
+            HeatRecovery::SimHeatRecovery(state,
+                                          thisHXCoil.HeatExchangerName,
+                                          FirstHVACIteration,
+                                          thisHXCoil.HeatExchangerIndex,
+                                          FanOpMode,
+                                          PartLoadRatio,
+                                          HXUnitOn,
+                                          CompanionCoilIndexNum,
+                                          _,
+                                          EconomizerFlag,
+                                          _,
+                                          thisHXCoil.CoolingCoilType_Num);
 
-            if (state.dataHVACAssistedCC->HXAssistedCoil(HXAssistedCoilNum).CoolingCoilType_Num == CoilDX_Cooling) {
+            if (thisHXCoil.CoolingCoilType_Num == DataHVACGlobals::CoilDX_Cooling) {
 
-                int coolingCoilIndex = state.dataHVACAssistedCC->HXAssistedCoil(HXAssistedCoilNum).CoolingCoilIndex;
+                int coolingCoilIndex = thisHXCoil.CoolingCoilIndex;
 
                 int mSingleMode = state.dataCoilCooingDX->coilCoolingDXs[coolingCoilIndex].getNumModes();
                 bool singleMode = (mSingleMode == 1);
 
                 Real64 mCoolingSpeedNum = state.dataCoilCooingDX->coilCoolingDXs[coolingCoilIndex]
                                               .performance.normalMode.speeds.size(); // used the same for the original variable speed coil
 
-                Real64 CoilPLR = 1.0;
-                int mControlType = 2;
-                if (mControlType == 2) {
```

Comment:
[src/EnergyPlus/HVACHXAssistedCoolingCoil.cc:1111]:(style),[knownConditionTrueFalse],The comparison 'mControlType == 2' is always true.